packages/rest/src/lib/utils/utils.ts

```diff
@@ -33,7 +33,7 @@ function serializeSearchParam(value: unknown): string | null {
  * @param options - The options to use
  * @returns A populated URLSearchParams instance
  */
-export function makeURLSearchParams<T extends object>(options?: Readonly<T>) {
+export function makeURLSearchParams<Params extends object>(options?: Readonly<Params>) {
```

Comment:
Not going to block on this cause I don't know what else to call it, but calling this Params makes me feel like it wants to be passed some variant of URLSearchParams and not any generic (pun not intended) interface. It's also mostly for inference to work better since satisfies exists but ya know..

Reply:
How about simply `OptionsType`?

Reply:
Sure, why not. I never really liked this generic anyways, but it does make stuff play better together, so that seems pretty clear to describe what's going on.

Reply:
Since `satisfies` is now a thing (it wasn't when I wrote it), can't we drop this generic altogether and change the input to `object`?

Reply:
Funnily enough I think I do use satisfies in places where I call this in app code rn lol. I was wondering why it wasn't that way, but I thought satisfies was out when I looked at the blame time. Guess not. I'd say just leave this for now then cause that will affect a lot of stuff very out of scope for this PR but can be removed soon (debating whether that's major or not).

Reply:
That's fine by me.

As for the breakability, I think types are always semver-patch? They are in the main package, after all.

Reply:
Dropping the generic isn't really sem patch... Changing its type to fix a bug is tho